### ISSUE TITLE
Regex.Matches no longer returns the empty trailing match for the .* pattern

### ISSUE THREAD

**Issue by slozier** [dotnet/runtime#42392]
### Description

Regex.Matches no longer returns the empty trailing match for the .* pattern. I expected the following to succeed:
```C#
Debug.Assert(new Regex(".*").Matches("abc").Select(x => x.Value).SequenceEqual(new List<string> { "abc", "" }));
```

I'm not sure if this should be considered a bug or if it's an intentional change. For comparison, Python also include the empty entry in its result set (`re.findall(".*", "") == ['abc', '']`).

### Configuration

.NET SDK (reflecting any global.json):
Version: 5.0.100-rc.1.20452.10
Commit: 473d1b592e

Runtime Environment:
OS Name: Windows
OS Version: 10.0.19042
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-rc.1.20452.10\

Host (useful for support):
Version: 5.0.0-rc.1.20451.14
Commit: 38017c3

### Regression?

This works in .NET Core 3.1 and .NET Framework.

**Comment by stephentoub** [comment]
This is the same root cause as https://github.com/dotnet/runtime/issues/42390.  Fixing that will fix this, too.

**Comment by jeffhandley** [comment]
Fixed in #42408 (master) and #42409 (release/5.0)
